dot/rpc/modules/author_test.go

```diff
@@ -14,173 +14,145 @@ import (
-	mockHasKey.Run(func(args mock.Arguments) {
-		pubKeyHex := args.Get(0).(string)
-		keyType := args.Get(1).(string)
-
-		ok, err := keystore.HasKey(pubKeyHex, keyType, globalStore.Acco)
-		mockHasKey.ReturnArguments = []interface{}{ok, err}
-	})
-
-	keys := "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d34309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc3852042602634309a9d2a24213896ff06895db16aade8b6502f3a71cf56374cc38520426026"
-	runtimeInstance := wasmer.NewTestInstance(t, runtime.NODE_RUNTIME)
-
-	decodeSessionKeysMock := coremockapi.On("DecodeSessionKeys", mock.AnythingOfType("[]uint8"))
-	decodeSessionKeysMock.Run(func(args mock.Arguments) {
-		b := args.Get(0).([]byte)
-		dec, err := runtimeInstance.DecodeSessionKeys(b)
-		decodeSessionKeysMock.ReturnArguments = []interface{}{dec, err}
-	})
-
-	module := &AuthorModule{
-		coreAPI: coremockapi,
-		logger:  log.New(log.SetWriter(io.Discard)),
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+		expErr  error
+		exp     HasSessionKeyResponse
+	}{
+		{
+			name: "Empty Request",
+			fields: fields{
+				authorModule: NewAuthorModule(log.New(log.SetWriter(io.Discard)), coreMockAPIDecodeErr, nil),
+			},
+			args: args{
+				req: &HasSessionKeyRequest{},
+			},
+			exp:     false,
+			wantErr: true,
+			expErr:  errors.New("invalid string"),
+		},
+		{
+			name: "decodeSessionKeys err",
+			fields: fields{
+				authorModule: NewAuthorModule(log.New(log.SetWriter(io.Discard)), coreMockAPIUnmarshalErr, nil),
+			},
+			args: args{
+				req: &HasSessionKeyRequest{"0x01"},
+			},
+			exp:     false,
+			wantErr: true,
+			expErr:  errors.New("unsupported Option value: 4, bytes: [1]"),
+		},
+		{
+			name: "happy path",
+			fields: fields{
+				authorModule: NewAuthorModule(log.New(log.SetWriter(io.Discard)), coreMockAPIOk, nil),
+			},
+			args: args{
+				req: &HasSessionKeyRequest{testReq},
+			},
+			exp:     true,
+			wantErr: false,
+		},
+		{
+			name: "doesnt have key",
+			fields: fields{
+				authorModule: NewAuthorModule(log.New(log.SetWriter(io.Discard)), coreMockAPIErr, nil),
+			},
+			args: args{
+				req: &HasSessionKeyRequest{testReq},
+			},
+			exp:     false,
+			wantErr: true,
+			expErr:  errors.New("HasKey err"),
+		},
+		{
+			name: "Empty decodedKeys",
+			fields: fields{
+				authorModule: NewAuthorModule(log.New(log.SetWriter(io.Discard)), coreMockAPIInvalidDec, nil),
+			},
+			args: args{
+				req: &HasSessionKeyRequest{testReq},
+			},
+			exp:     false,
+			wantErr: false,
+		},
 	}
-
-	req := &HasSessionKeyRequest{
-		PublicKeys: keys,
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			am := tt.fields.authorModule
+			err := am.HasSessionKeys(tt.args.r, tt.args.req, &tt.args.res)
+			if tt.wantErr {
+				require.Error(t, err)
+				assert.EqualError(t, err, tt.expErr.Error())
```

Comment:
nit no need for both since `assert.EqualError` takes care of it

```suggestion
				assert.EqualError(t, err, tt.expErr.Error())
```